client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -114,7 +118,22 @@ class CelebornShuffleReader[K, C](
         var inputStream: CelebornInputStream = streams.get(partitionId)
         while (inputStream == null) {
           if (exceptionRef.get() != null) {
-            throw exceptionRef.get()
+            exceptionRef.get() match {
+              case cioe: CelebornIOException =>
+                if (handle.throwsFetchFailure &&
+                  shuffleClient.reportShuffleFetchFailure(handle.shuffleId, shuffleId)) {
+                  throw new FetchFailedException(
+                    null,
+                    handle.shuffleId,
+                    -1,
+                    -1,
+                    partitionId,
+                    SparkUtils.FETCH_FAILURE_ERROR_MSG,
+                    cioe)
```

Comment:
Please add a note to `reportShuffleFetchFailure` that it must be a sync call.
Unlike in 'regular' spark codepaths, where `FetchFailedException` triggers special handling, here we are relying on `reportShuffleFetchFailure` to do the same.

Reply:
sure, will do